pkg/cloud/aws/services/ec2/instances.go

```diff
@@ -346,11 +362,26 @@ func (s *Service) createInstance(machine *actuators.MachineScope, bootstrapToken
 				),
 			),
 		)
+
+		if amazonVPCCNIEnabled {
+			kubeadm.SetJoinConfigurationOptions(
+				kubeadmConfig,
+				kubeadm.WithJoinNodeRegistrationOptions(
+					kubeadm.SetNodeRegistrationOptions(
+						&kubeadmConfig.NodeRegistration,
+						kubeadm.WithKubeletExtraArgs(map[string]string{"max-pods": strconv.Itoa(ENILimit)}),
```

Comment:
Should max-pods also be set NewInitControlPlane as well and NewJoinControlPlane as well?

Reply:
Yes, and I would suggest a `min(userValue, strconv.Itoa(ENILimit))` strategy if the incoming config already includes a `max-pods`.

We're also missing a few flags that we should use a default if-not-set for:

- `--cni-conf-dir=/etc/cni/net.d`
- `--cni-bin-dir=/opt/cni/bin`

(to match the manifest you provided at the top)

Reply:
> Should max-pods also be set NewInitControlPlane as well and NewJoinControlPlane as well?

many pods in master node work the host network is true. so I remove the max-pods. Maybe we need to set a correct value of this parameter.

Reply:
> Yes, and I would suggest a `min(userValue, strconv.Itoa(ENILimit))` strategy if the incoming config already includes a `max-pods`.
> 
> We're also missing a few flags that we should use a default if-not-set for:
> 
> * `--cni-conf-dir=/etc/cni/net.d`
> * `--cni-bin-dir=/opt/cni/bin`
> 
> (to match the manifest you provided at the top)

got, We will choose a smaller value if the user set the max-pods.